File: bench/arith.js
// The JavaScript twin of arith.sl. Every JavaScript number is a double, so this is real arithmetic
// where slate's and Lua's is integer -- the answer, 99,999,980,000,000, is well inside 2^53, so it
// is exact and prints the same.

function run() {
    let total = 0;
    let i = 0;

    while (i < 10000000) {
        total = total + i * 2 - 1;
        i = i + 1;
    }

    return total;
}

console.log(run());

File: bench/arith.lua
-- The Lua twin of arith.sl. Lua 5.5 has a 64-bit integer subtype, so this stays integer arithmetic
-- exactly as slate's does and the answer prints without a decimal point.

local function run()
    local total, i = 0, 0

    while i < 10000000 do
        total = total + i * 2 - 1
        i = i + 1
    end

    return total
end

print(run())
